presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -472,6 +472,70 @@ public void testLineLocatePoint()
         assertInvalidFunction("line_locate_point(ST_GeometryFromText('LINESTRING (0 0, 0 1, 2 1)'), ST_GeometryFromText('POLYGON ((1 1, 1 4, 4 4, 4 1))'))", "Second argument to line_locate_point must be a Point. Got: Polygon");
     }
 
+    @Test
+    public void testLineInterpolatePoint()
+    {
+        assertFunction("ST_AsText(line_interpolate_point(ST_GeometryFromText('LINESTRING EMPTY'), 0.5))", VARCHAR, null);
+
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.0, 0.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.1, 1.0, 1.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.05, 0.5, 0.5);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.4, 4.000000000000001, 4.000000000000001);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 01.0, 10.0, 10.0);
+
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.0, 0.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.1, 0.1, 0.1);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.05, 0.05, 0.05);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.4, 0.4, 0.4);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 1.0, 1.0, 1.0);
+
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 0, 1 9)", 0.0, 0.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 0, 1 9)", 0.1, 1.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 0, 1 9)", 0.5, 1.0, 4.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 0, 1 9)", 1.0, 1.0, 9.0);
+
+        assertInvalidFunction("line_interpolate_point(ST_GeometryFromText('LINESTRING (0 0, 1 0, 1 9)'), -0.5)", INVALID_FUNCTION_ARGUMENT, "fraction must be between 0 and 1");
+        assertInvalidFunction("line_interpolate_point(ST_GeometryFromText('LINESTRING (0 0, 1 0, 1 9)'), 2.0)", INVALID_FUNCTION_ARGUMENT, "fraction must be between 0 and 1");
+        assertInvalidFunction("line_interpolate_point(ST_GeometryFromText('POLYGON ((0 0, 1 1, 0 1, 1 0, 0 0))'), 0.2)", INVALID_FUNCTION_ARGUMENT, "line_interpolate_point only applies to LINE_STRING. Input type is: POLYGON");
+    }
+
+    private void assertLineInterpolatePoint(String wkt, double fraction, double expectedX, double expectedY)
```

Comment:
The tests would be easier to read if instead we passed in an expected wkt. So, instead of:

```
assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.4, 0.4, 0.4);

```

we could do:

```
assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.4, "POINT (0.4 0.4)");
```

Reply:
Also, move this method to the bottom of the class near `assertLineInterpolatePoints`